### PR TITLE
Introduce reprs for some of the base objects

### PR DESCRIPTION
**Pull request by rgbkrk**
This introduces ipython / Jupyter representations for Python objects to allow easier viewing in a notebook.

For instance, running this code will display all the charts from `lida.visualize`

> ```python
> i = 0
> library = "seaborn"
> textgen_config = TextGenerationConfig(n=1, temperature=0.2, use_cache=True)
> charts = lida.visualize(
>     summary=summary, goal=goals[i], textgen_config=textgen_config, library=library
> )
> for chart in charts:
>     display(chart)
> ```
> 
> image

and you can also just the `chart` be the execution result for the cell:

> ```python
> library = "seaborn"
> textgen_config = TextGenerationConfig(n=1, temperature=0.2, use_cache=True)
> charts = lida.visualize(
>     summary=summary, goal=goals[i], textgen_config=textgen_config, library=library
> )
> charts[0]
> ```
> 
> image

Additionally, I did the same for goals:

image

Working on updating the notebook example as well once my machine has the infographics dependencies working well.

### CODE REVIEW

**Comment by rgbkrk** [comment]
@microsoft-github-policy-service agree company="Noteable"

File: lida/datamodel.py [hunk 1]
```diff
@@ -24,21 +26,35 @@ class CompletionResult:
...
 @dataclass
 class Goal:
     """A visualization goal"""
+
     index: int
     question: str
     visualization: str
     rationale: str
 
+    def _repr_markdown_(self):
```
Comment (victordibia) [inline lida/datamodel.py:42]: Thanks for the _repr_'s. 
I learned a few new things here @rgbkrk !!

File: lida/components/executor.py [hunk 2]
```diff
@@ -101,12 +100,14 @@ def execute(
         #     raise Exception(
         #         "Permission to execute code not granted. Please set the environment variable LIDA_ALLOW_CODE_EVAL to '1' to allow code execution.")
 
+        if isinstance(summary, dict):
```
Comment (victordibia) [inline lida/components/executor.py:103]: Good catch here.

**Review by victordibia — APPROVED** [review]

File: lida/datamodel.py (hunk 1 above)
Comment (rgbkrk) [reply lida/datamodel.py:42]: Always happy to sprinkle on some ipython goodness. 😄

**Pull request merged**